exercise/sitemaps.py

```diff
@@ -0,0 +1,53 @@
+from django.contrib import sitemaps
+from django.utils import timezone
+
+from course.models import CourseInstance
+from .models import BaseExercise, CourseChapter
+
+
+class BaseExerciseSitemap(sitemaps.Sitemap):
+    priority = 0.2
+    changefreq = 'daily'
+
+    def items(self):
+        return BaseExercise.objects.filter(
+            course_module__course_instance__view_content_to=CourseInstance.VIEW_ACCESS.PUBLIC,
+            course_module__opening_time__lte=timezone.now(),
+        )
+
+    def location(self, item):
+        return item.get_display_url()
+
+
+class CourseChapterSitemap(sitemaps.Sitemap):
+    priority = 1.0
+    changefreq = 'daily'
+
+    def items(self):
+        return CourseChapter.objects.filter(
```

Comment:
I forgot to say earlier that some chapters or exercises can actually be hidden or behind the authentication. The sitemap should only include chapters or exercises that have the status "ready" or "unlisted": 
https://github.com/apluslms/a-plus/blob/c6044f5b785336966e504c5d163d2a786edcdb6c/exercise/exercise_models.py#L69-L70

In addition, if the chapter or exercise requires the user to login, then it should not be listed in the sitemap: 
https://github.com/apluslms/a-plus/blob/c6044f5b785336966e504c5d163d2a786edcdb6c/exercise/exercise_models.py#L84
If I remember correctly, then only `audience == COURSE_AUDIENCE` could be public. It is public if the course is public. The other audience values always require the user to login.

https://github.com/apluslms/a-plus/blob/c6044f5b785336966e504c5d163d2a786edcdb6c/exercise/permissions.py#L39

Reply:
Ok, I'll add more filtering to the course chapter and exercise sitemaps.